draft-aegis-mimi-arch-latest.md

```diff
@@ -170,127 +172,133 @@ provider. It has the following properties:
 
 ## X.509 Identity Provider
 
-The X.509 identify provider allows for a hierarchical identity that is based
-upon a certificate chain. x509 is the only credential type supported.
+The X.509 identify provider allows for hierarchical identities based on
+certificate chains. The provider is configured via the `X509Parameters` struct. 
+Besides clients, depending on how the provider is configured it may also support
+accounts and/or domains. Conceptually, an account is a collection of clients
+(usually belonging to one user) and a domain is a collection of accounts
+(e.g. belonging to a particular organization).
+
+The only credential supported by the X509 Identity Provider are `x509` 
+credentials which consist of a chain of 1 or more X.509 certificates. 
 Certificate chains MUST be validated according to the rules in RFC 5280 using
 the trust roots agreed upon by the (TODO: Some sort of extension dealing with
 trust root negotiation).
 
 ~~~
 struct {
-   uint64 msg_timestamp;
+    uint64 msg_timestamp;
 } X509ApplicationContext;
 
 struct {
-    X509IdentityRange range;
+    uint8 start;
+    uint8 end<0...255>;
+} X509EndpointHandleRange
```

Comment:
"Endpoint" is confusing if in the same doc we have REST endpoints... Maybe session for the lack of a better word?